File: frontend/src/features/onboarding/schemas.ts
// 온보딩 도메인 Zod 스키마 — BE GET /api/v1/users/me/onboarding 응답 검증
import { z } from "zod/v4";

export const onboardingResponseSchema = z.object({
  step: z.number().int().min(0).max(4),
  totalSteps: z.literal(4),
  onboardedAt: z.string().datetime().nullable(),
  isCompleted: z.boolean(),
});

export type OnboardingResponse = z.infer<typeof onboardingResponseSchema>;

File: apps/api/src/onboarding/schemas.py
# Onboarding 도메인 — Pydantic 스키마 (server-side persistent step tracker)
from datetime import datetime
from enum import IntEnum

from pydantic import BaseModel, ConfigDict, Field


class OnboardingStep(IntEnum):
    """User.onboarding_step lifecycle. Sprint 22 OBN-02."""

    NOT_STARTED = 0
    WORKSPACE_CREATED = 1
    FIRST_PROJECT = 2
    FIRST_MEETING = 3
    FIRST_RAG = 4


class OnboardingResponse(BaseModel):
    """GET /api/v1/users/me/onboarding 응답."""

    model_config = ConfigDict(populate_by_name=True)

    step: int = Field(..., alias="step")
    total_steps: int = Field(4, alias="totalSteps")
    onboarded_at: datetime | None = Field(None, alias="onboardedAt")
    is_completed: bool = Field(False, alias="isCompleted")
